### ISSUE TITLE
Do not enforce on disk filename for models

### ISSUE THREAD

**Issue by tazlin** [Haidra-Org/hordelib#52]
Ultimately, when it comes to worker users already having models downloaded, the expectation that the file is named is a certain way is a sore point. Looking at only the SHA256 sum should be sufficient.

**Comment by db0** [comment]
Wouldn't that mean we have to make a filemap of some sort?

**Comment by tazlin** [comment]
Yes. I feel up to the challenge of engineering an appropriate system if time ever permits. The `horde_model_reference` library was designed specifically with this sort of flexibility in mind.

**Comment by tazlin** [comment]
Ideally, this would fit into the same system I envision supporting N model folders, and having user customizable mapping of the variety I think you're hinting at, allowing cross A1111/Comfy/Horde use of models.

**Comment by tazlin** [comment]
See https://github.com/orgs/Haidra-Org/projects/6 for a higher level picture of some of the things I would one day like to see.
